smarts/core/waypoints.py

```diff
@@ -356,6 +367,21 @@ def _waypoints_starting_at_waypoint(
 
         return [self._equally_spaced_path(path, point) for path in waypoint_paths]
 
+    @staticmethod
+    def _inplace_unwrap(wp_array):
+        ## minor optimization hack adapted from
+        ##  https://github.com/numpy/numpy/blob/v1.20.0/numpy/lib/function_base.py#L1492-L1546
+        ## to avoid unnecessary (slow) np array copy
+        ## (as seen in profiling).
+        p = np.asarray(wp_array)
+        dd = np.subtract(p[1:], p[:-1])
+        ddmod = np.mod(dd + math.pi, 2 * math.pi) - math.pi
+        np.copyto(ddmod, math.pi, where=(ddmod == -math.pi) & (dd > 0))
+        ph_correct = ddmod - dd
+        np.copyto(ph_correct, 0, where=abs(dd) < math.pi)
+        p[1:] += ph_correct.cumsum(axis=-1)
+        return p
```

Comment:
I believe `_inplace_unwrap()` can be placed as a utility function outside the `Waypoints` class, as it is not necessarily tied to waypoints.

Reply:
done.

Reply:
done.